services/user_service.js

```diff
@@ -1,15 +1,41 @@
-const { getPool } = require('../db');
-const pool = getPool();
-const utils = require('../utils');
+//const { getPool } = require('../db');
+//const pool = getPool();
+//const utils = require('../utils');
+const rolesService = require('./roles_service');
+const repo = require('../repositories/user_repository');
 
-async function assignRoleToUser(userId, roleName){
-    const roleId = utils.getRole(roleName);
+async function assignRoleToUser(userId, roleName) {
     try {
-        await pool.query("INSERT INTO user_roles (user_id, role_id) VALUES ($1, $2)",
-            [userId, roleId]);
+        const roleId = rolesService.getRole(roleName);
+      await repo.assignRoleToUser(userId, roleId);
     } catch (err) {
         throw new Error(`Failed to assign role to user ${userId}:  ${err.message}`);
     }
 }
+async function getAllUsers() {
 
-module.exports = { assignRoleToUser };
\ No newline at end of file
+    return await repo.getAllUsers();
+}
+async function createNewUser(name, email, about, languages) {
+    return await repo.createNewUser(name, email, about, languages);
+}
+async function getUserById(id) {
+    let userInfo = await repo.getUserInfoById(id) || {};
+
+    if (!userInfo) {
+
+        const roles = await repo.getUserRolesById(id);
+        userInfo.roles = roles.length > 0 ? roles : [];
```

Comment:
if roles.length is not greater than 0, isn't it already an empty array?

Reply:
yes looks like it, also roles.length is already checking if length of the array is > 0, so usually it's written like roles.length ? <true> : <false>